clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/DefaultInputDataInfoProvider.java

```diff
@@ -357,7 +357,8 @@ protected Pair<Schema, Schema> getAvroFileHeader(FileSystem fs, Path path, boole
         if (!pushJobSetting.isIncrementalPush) {
           if (!pushJobSetting.useMapperToBuildDict) {
             /** If dictionary compression is enabled for version, read the records to get training samples */
-            if (pushJobSetting.storeCompressionStrategy == CompressionStrategy.ZSTD_WITH_DICT) {
+            if (pushJobSetting.storeCompressionStrategy == CompressionStrategy.ZSTD_WITH_DICT
+                || pushJobSetting.compressionMetricCollectionEnabled) {
```

Comment:
similar check needed in `getMetadataFromSequenceFile`

Reply:
Hmm, interesting. Good catch. Wonder why the test didn't catch this